### DISCUSSION TITLE
Benchmarking

### DISCUSSION THREAD

**Discussion by gregdenay** [bio-raum/FooDMe2#37] [General]
Benchmarking is implemented. We can now autmatically test how good a tool /parameters work.
To use: `--ground_truth` must provide the path to tsv file with columns `sample`, `taxid`,  `proportion`.
Additionnally one can define/modify a max rank for matching truth and prediction (`--benchmark_rank`) and a minimum proportion (`--benchmark_cutoff`)

The output is still a bit rough (JSON), maybe we can integrate it to Multiqc in the long run

**Comment by marchoeppner** [comment]
I was just checking the benchmark results from the documentation with my own results. There are some discrepencies. Could you check on your end to see if this actually matches? In principle, if we use the same database and pipeline version, the numbers should be the same - and the files should be identical on the byte level. 

```bash
nextflow run /home/mhoeppner/git/FooDMe2/main.nf -profile lsh,dobrovolny_benchmark --input ../data/Foodme/samples.tsv
``` 
where samples.tsv are the 79 samples from the FooDMe paper. Benchmark results as follows:

TP: 524
FN: 5
FP: 27

md5sum: 2dcc4337b0bc5361f0b0b499a1ccace5  results/reports/Dobrovolny_benchmark_benchmark.xlsx

And the md5sums for the reads:

[md5sums.txt](https://github.com/user-attachments/files/17054523/md5sums.txt)

**Reply by gregdenay** [reply]
So I just checked in several runs, I get a different md5 every run. The content of the table howver is the same. I assume excel keeps some metadata in the file, which are different every run.
I count following in the excel file:
FN: 19
FP: 31
TP: 524

**Reply by marchoeppner** [reply]
Good point about excel; probably better to dump to csv first. In any case, since the numbers for you and me are different, this begs the uncomfortable question of "why" ;)

**Reply by gregdenay** [reply]
I checked the fastq md5 too, that's all the same. 
Here is the excel I got with a commit from today
[Dobrovolny_benchmark_benchmark.xlsx](https://github.com/user-attachments/files/17057600/Dobrovolny_benchmark_benchmark.xlsx)

**Reply by marchoeppner** [reply]
Well, the good news is that your file and mine are identical (md5sum after csv conversion). The bad news, one of us can't count, and I have a suspicion it might be me ;)

**Comment by marchoeppner** [comment]
Unrelated question - the config for the benchmark currently sets max_cpus = 6. Is this intentional, or just a copy/paste?

The limit on the test profile is to make it possible to run the test on the github runner. For a local benchmark run, you probably want as many cpus as possible (79 samples, after all).

**Reply by gregdenay** [reply]
no it's an error, should be removed

**Reply by marchoeppner** [reply]
Fixed.
